File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp; // value -> index

        for(int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];

            // If the complement exists, return its index and current index
            if(mp.find(diff) != mp.end()) {
                return { mp[diff], i };
            }

            // Store the number with its index
            mp[nums[i]] = i;
        }

        return {}; // not really needed since one solution always exists
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // int index0 = 0, index1 = 0;
        int[] retArr = new int[2] ;
        for (int i=0; i < nums.length; i++){
            // if(nums[i]>target){continue;}
            for (int j=i+1;j<nums.length;j++){
                // if(nums[j]>target){continue;}
                if(nums[i]+nums[j] == target){
                    retArr[0] = i;
                    retArr[1] = j;
                }
            }
        }
        return retArr;
    }
}